src/main/java/frc/robot/subsystems/DriveSystem.java

```diff
@@ -332,31 +332,69 @@ public CommandBase rotateToAngle(Rotation2d angle) {
         ChassisSpeeds rotationVel = new ChassisSpeeds(0, 0, nextVel);
         DifferentialDriveWheelSpeeds wheelSpeeds = kinematics.toWheelSpeeds(rotationVel);
 
-        // clamp wheel speeds to max velocity
-        double left = MathUtil.clamp(wheelSpeeds.leftMetersPerSecond, -MAX_SPEED, MAX_SPEED);
-        double right = MathUtil.clamp(wheelSpeeds.rightMetersPerSecond, -MAX_SPEED, MAX_SPEED);
-
-        // apply drivetrain speeds to drive pid controllers
-        leftController.setReference(left, ControlType.kVelocity);
-        rightController.setReference(right, ControlType.kVelocity);
+        setDrivePIDControllers(wheelSpeeds);
       },
       // runs once at end of command 
       () -> {
-        rotateController.reset();
+        finishRotation();
+      }
+    ).until(
+      // returns true if robot is at end angle
+      rotateController::atSetpoint
+    );
+  }
 
-        // stop motors
-        leftController.setReference(0, ControlType.kVelocity);
-        rightController.setReference(0, ControlType.kVelocity);
+  public CommandBase autoBalance() {
+    
+    return runEnd(
+      // runs repeatedly until end of command
+      () -> {
+        // radians
+        double currAngle = Math.toRadians(gyro.getRoll());
 
-        frontLeft.stopMotor();
-        frontRight.stopMotor();
+        double endAngle = 0;
+
+        // rad/s ????
+        double nextVel = rotateController.calculate(currAngle, endAngle);
+
+        // convert radial velocity to drivetrain speeds
+        ChassisSpeeds balanceVal = new ChassisSpeeds(nextVel, 0, 0);
+        DifferentialDriveWheelSpeeds wheelSpeeds = kinematics.toWheelSpeeds(balanceVal);
+
+        setDrivePIDControllers(wheelSpeeds);
+      },
+      // runs once at end of command 
+      () -> {
+        finishRotation();
       }
     ).until(
       // returns true if robot is at end angle
       rotateController::atSetpoint
     );
   }
 
+  private void setDrivePIDControllers(DifferentialDriveWheelSpeeds wheelSpeeds) {
+    // clamp wheel speeds to max velocity
+    double left = MathUtil.clamp(wheelSpeeds.leftMetersPerSecond, -MAX_SPEED, MAX_SPEED);
+    double right = MathUtil.clamp(wheelSpeeds.rightMetersPerSecond, -MAX_SPEED, MAX_SPEED);
+
+    // apply drivetrain speeds to drive pid controllers
+    leftController.setReference(left, ControlType.kVelocity);
+    rightController.setReference(right, ControlType.kVelocity);
+  }
+
+  private void finishRotation() {
+    rotateController.reset();
+
+    // stop motors
+    leftController.setReference(0, ControlType.kVelocity);
+    rightController.setReference(0, ControlType.kVelocity);
+
+    frontLeft.stopMotor();
+    frontRight.stopMotor();
+  }
```

Comment:
remove this method: since balance shouldn't be using rotatecontroller the end code will be different